pkg/chart/repo/repo.go

```diff
@@ -61,6 +61,13 @@ func (r *Repo) RefreshIndex() (*repo.IndexFile, error) {
 
 	data, err := r.fetcher(indexURL)
 	if err != nil {
+		_, cacheErr := r.cache.Fetch("index.yaml")
+		if cacheErr != nil {
```

Comment:
Do we care about cacheErr contents? It seems it's lost after a non-nil check

Reply:
The error that we are printing to logs is the response code from the server when trying to fetch the chart (for example: "unsuccessful response code: 500 Internal Server Error (500)"). cacheErr will hold the error for trying to fetch from cache (for example: "open /tmp/chart-cache/https-charts.prod.booking.com-charts/index.yaml: no such file or directory")
I can add cacheErr if you think it should be there as well

Reply:
I see your point. It seems to be a composite error: we got err and cacheErr, both worth reporting. I might suggest using shippererrors.MultiError in that case so we don't miss the actual err reason contained in err.